106-self-healing-feature-in-operator.md

```diff
@@ -0,0 +1,295 @@
+
+#### Notifiers in Cruise Control
+
+Whenever anomalies are detected, Cruise Control provides the ability to notify the user regarding the detected anomalies using optional notifier classes.
+The notification sent by these classes increases the visibility of the operations that are taken by Cruise Control.
+The notifier class used by Cruise Control is configurable and custom notifiers can be used by setting the `anomaly.notifier.class` property.
+The notifier class returns the `action` that is going to be taken on the flagged anomaly.
+These actions have three types:
+* FIX - Start the anomaly fix
+* CHECK - Delay the anomaly fix
+* IGNORE - Ignore the anomaly fix
+
+The default `NoopNotifer` always sets the notifier action as `IGNORE`, which  means that the detected anomaly will be silently ignored and no notification is sent to the user.
+
+Cruise Control also provides [custom notifiers](https://github.com/linkedin/cruise-control/wiki/Configure-notifications) like Slack Notifier, Alerta Notifier etc. for notifying users regarding the anomalies. There are multiple other [self-healing notifier](https://github.com/linkedin/cruise-control/wiki/Configurations#selfhealingnotifier-configurations) related configurations you can use to make notifiers more efficient as per the use case.
+
+#### Self Healing
+
+If self-healing is enabled, then an action is returned by the notifier to would decide whether the anomaly should be fixed or not.
+If the notifier has returned `FIX` as the action then the classes which are responsible for resolving the anomaly would be called.
+Each detectable anomaly is handled by a specific detector class which then uses another remediation class to run a fix.
+For example, the `GoalViolations` class uses the `RebalanceRunnable` class, the `DiskFailure` class use the `RemoveDisksRunnable` class and so on.
+An optimization proposal will then be generated by these `Runnable` classes and that proposal will be applied on the cluster to fix the anomaly.
+In case the anomaly detected is unfixable for e.g. violated hard goals that cannot be fixed typically due to lack of physical hardware (insufficient number of racks to satisfy rack awareness, insufficient number of brokers to satisfy Replica Capacity Goal, or insufficient number of resources to satisfy resource capacity goals), the anomaly wouldn't be fixed and the Cruise Control logs will be updated with `self-healing is not possible due to unfixable goals` warning.
+
+### Proposal
+
+This proposal allows the users to have their cluster balanced automatically whenever an anomaly is detected by Cruise Control.
+To ensure that the operator drives the rebalance we will be leveraging the Cruise Control self-healing mechanism but not completely.
```

Comment:
We are not leveraging the self-healing at all, we are leveraging the anomaly detection only. We need to be careful about mentioning self-healing here. As said before people could get confused if we are using or not (which is the case) the CC self-healing feature.